js/coinflex.js

```diff
@@ -0,0 +1,1920 @@
+        //          "code": "710003",
+        //          "message": "FAILED sanity bound check as price (4.000) >  upper bound (3.439)",
+        //          "clientOrderId": "1651619024219000",
+        //          "price": "4.000",
+        //          "quantity": "3.0",
+        //          "side": "BUY",
+        //          "marketCode": "BAND-USD",
+        //          "timeInForce": "GTC",
+        //          "orderType": "LIMIT"
+        //     }
+        //
+        const data = this.safeValue (response, 'data', []);
+        const firstResult = this.safeValue (data, 0, {});
+        const success = this.safeValue (firstResult, 'success');
+        if (success === 'false') {
+            const message = this.safeString (firstResult, 'message');
+            const code = this.safeString (firstResult, 'code');
+            const body = this.id + ' ' + this.json (response);
+            this.throwExactlyMatchedException (this.exceptions['exact'], code, body);
+            this.throwBroadlyMatchedException (this.exceptions['broad'], message, body);
+            throw new ExchangeError (body);
+        }
+        return this.parseOrder (firstResult, market);
+    }
+
+    async cancelAllOrders (symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = undefined;
+        const request = {};
+        let method = 'privateDeleteV2CancelOrders';
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['marketCode'] = market['id'];
+            method = 'privateDeleteV2CancelOrdersMarketCode';
+        }
+        const response = await this[method] (this.extend (request, params));
+        //
+        //     {
+        //         "event": "orders",
+        //         "timestamp": "1651651758625",
+        //         "accountId": "38420",
+        //         "data": {
+        //             "marketCode": "XRP-USD",
+        //             "msg": "All open orders for the specified market have been queued for cancellation"
+        //         }
+        //     }
+        //
+        // Note: when fired without symbol, the 'data' object only contains a msg property with '{"msg":"All open orders for the account have been queued for cancellation"}'
+        // if there has been no orders pending, then data property will be null.
+        //
+        return response;
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        const [ finalPath, query ] = this.resolvePath (path, params);
+        let url = this.urls['api'][api] + '/' + finalPath;
+        let encodedParams = '';
+        if (Object.keys (query).length) {
+            encodedParams = this.urlencode (query);
+            if (method === 'GET') {
+                url += '?' + encodedParams;
+            }
+        }
+        if (api === 'private') {
+            this.checkRequiredCredentials ();
+            const nonce = this.nonce ();
```

Comment:
This nonce is an integer so in Python this breaks on the string concatenation below and in the header

Reply:
right, likewise previous issue with newlines, I've not completed building this rest for JS comlpetely, and havent tested other langs. i'll fix this now.